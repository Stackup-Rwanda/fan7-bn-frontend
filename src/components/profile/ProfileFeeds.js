import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import './styles/profileFeeds.scss';

class ProfileFeeds extends Component {
  render() {
    const { gender, dob, address } = this.props;

    return (
      <div className="feeds">
        <div className="feeds-card1">
          <h1 className="feeds-card1__title">Intro</h1>
          <div className="feeds-card1__content">
            {!gender && <span className="feeds-card1__content___icon" />}
            <p className="feeds-card1__content___text">
              {gender ? `Gender: ${gender}` : 'Gender'}
            </p>
          </div>
          <div className="feeds-card1__content">
            {!dob && <span className="feeds-card1__content___icon" />}
            <p className="feeds-card1__content___text">
              {dob ? `Birthday: ${moment(dob).format('L')}` : 'Birthday'}
            </p>
          </div>
          <div className="feeds-card1__content">
            {!address && <span className="feeds-card1__content___icon" />}
            <p className="feeds-card1__content___text">
              {address ? `Location: ${address}` : 'Current location'}
            </p>
          </div>
        </div>
        <div className="feeds-card2">
          <h1 className="feeds-card2__title">Fill Out Your Profile</h1>
          <p className="feeds-card2__desc">
            Add photos and info to your profile so people can find you easily and get to know you
            better!
          </p>
        </div>
      </div>
    );
  }
}

ProfileFeeds.propTypes = {
  gender: PropTypes.string.isRequired,
  dob: PropTypes.string.isRequired,
  address: PropTypes.string.isRequired,
};

export default ProfileFeeds;
