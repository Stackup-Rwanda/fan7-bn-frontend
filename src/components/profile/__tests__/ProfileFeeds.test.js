/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { mount, shallow } from 'enzyme';
import ProfileFeeds from '../ProfileFeeds';

describe('ProfileFeeds Component Test', () => {
  it('should return true if component exists', () => {
    const wrapper = mount(<ProfileFeeds />);

    expect(wrapper.find('.feeds').exists()).toBe(true);
  });

  it('should display appropriate text value', () => {
    const props = {
      gender: 'Male',
      dob: '1999-1-1',
    };
    const wrapper = shallow(<ProfileFeeds {...props} />);

    expect(
      wrapper
        .find('.feeds-card1__content___text')
        .first()
        .text()
    ).toEqual('Gender: Male');

    expect(
      wrapper
        .find('.feeds-card1__content___text')
        .at(1)
        .text()
    ).toEqual('Birthday: 01/01/1999');

    expect(
      wrapper
        .find('.feeds-card1__content___text')
        .last()
        .text()
    ).toEqual('Current location');
  });

  it('should display appropriate props', () => {
    const props = {
      address: 'Kigali',
    };
    const wrapper = shallow(<ProfileFeeds {...props} />);

    expect(
      wrapper
        .find('.feeds-card1__content___text')
        .first()
        .text()
    ).toEqual('Gender');

    expect(
      wrapper
        .find('.feeds-card1__content___text')
        .at(1)
        .text()
    ).toEqual('Birthday');

    expect(
      wrapper
        .find('.feeds-card1__content___text')
        .last()
        .text()
    ).toEqual('Location: Kigali');
  });
});

describe('<ProfileFeeds /> shallow rendering tests', () => {
  it('matches the snapshot', () => {
    const props = {
      gender: 'Male',
      dob: '1996-5-6',
      location: 'Kigali',
    };
    const tree = shallow(<ProfileFeeds {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
