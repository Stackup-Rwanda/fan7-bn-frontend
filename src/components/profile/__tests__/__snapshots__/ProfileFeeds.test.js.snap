// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileFeeds /> shallow rendering tests matches the snapshot 1`] = `
<div
  className="feeds"
>
  <div
    className="feeds-card1"
  >
    <h1
      className="feeds-card1__title"
    >
      Intro
    </h1>
    <div
      className="feeds-card1__content"
    >
      <p
        className="feeds-card1__content___text"
      >
        Gender: Male
      </p>
    </div>
    <div
      className="feeds-card1__content"
    >
      <p
        className="feeds-card1__content___text"
      >
        Birthday: 05/06/1996
      </p>
    </div>
    <div
      className="feeds-card1__content"
    >
      <span
        className="feeds-card1__content___icon"
      />
      <p
        className="feeds-card1__content___text"
      >
        Current location
      </p>
    </div>
  </div>
  <div
    className="feeds-card2"
  >
    <h1
      className="feeds-card2__title"
    >
      Fill Out Your Profile
    </h1>
    <p
      className="feeds-card2__desc"
    >
      Add photos and info to your profile so people can find you easily and get to know you better!
    </p>
  </div>
</div>
`;
